### PR TITLE
Documentation changes from ntoll

### PR DESCRIPTION
**Pull request by ntoll**
@dpgeorge ok... so trying to work around GitHub here... I believe if you rebase against the branch that's the HEAD fork in this PR you should get my documentation changes. Fingers crossed.

### CODE REVIEW

File: docs/spi.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+   Read at most ``nbytes``. Returns what was read.
+
+.. method:: spi.write(buffer)
+
+   Write the ``buffer`` of bytes to the bus.
+
+.. method:: spi.write_readinto(out, in)
+
+   Write the ``out`` buffer to the bus and read any response into the ``in``
+   buffer. The length of the buffers should be the same. The buggers can be
```
Comment (dpgeorge) [inline docs/spi.rst]: @ntoll I don't think the 3rd-last word in this sentence would pass BBC review :) :) :)

**Comment by dpgeorge** [comment]
I will try.

Shouldn't we also add documentation for help(spi) etc?

File: docs/spi.rst (hunk 1 above)
Comment (ntoll) [inline docs/spi.rst]: Freudian slip... will correct. :-)

**Comment by ntoll** [comment]
Yes we should add `help(spi)` based helpful  fragments.

**Comment by dpgeorge** [comment]
Nope, rebasing didn't work.

Cant' we just merge my SPI implementation, and then your docs?  We can merge rst docs and then help() docs in separate PRs.  I don't see a problem with that.

**Comment by ntoll** [comment]
+1 on that... much of the frustrations from this weekend were git related. Let's keep it simple.

**Pull request closed without merging**